### ISSUE TITLE
SyntaxError: Unexpected token B in JSON

### ISSUE THREAD

**Issue by shargreaves-millmead** [understrap/understrap#2007]
Hi

Using understrap 1.1.0 and a child theme, have been developing a theme with no problems using sass.
On friday it was working and running fine, Monday morning came around and as soon as I ran npm run watch, it gave me an error.
I have tried to solve it but I am getting nowhere fast.

`created js/child-theme.js in 1.1s

> understrap-child@1.1.0 js-minify
> terser --config-file src/build/terser.config.json --output js/child-theme.min.js js/child-theme.js

> understrap-child@1.1.0 css-postcss
> postcss --config src/build/postcss.config.js --replace "css/*.css" "!css/*.rtl*.css" "!css/*.min.css"

[nodemon] clean exit - waiting for changes before restart
SyntaxError: Unexpected token B in JSON at position 26064
    at JSON.parse (<anonymous>)
    at Object.parseSourceMapInput (/Users/stephen/Local Sites/test-1/app/public/wp-content/themes/understrap-child/node_modules/source-map-js/lib/util.js:539:15)
    at new SourceMapConsumer (/Users/stephen/Local Sites/test-1/app/public/wp-content/themes/understrap-child/node_modules/source-map-js/lib/source-map-consumer.js:17:22)
    at PreviousMap.consumer (/Users/stephen/Local Sites/test-1/app/public/wp-content/themes/understrap-child/node_modules/postcss/lib/previous-map.js:33:28)
    at new Input (/Users/stephen/Local Sites/test-1/app/public/wp-content/themes/understrap-child/node_modules/postcss/lib/input.js:52:24)
    at parse (/Users/stephen/Local Sites/test-1/app/public/wp-content/themes/understrap-child/node_modules/postcss/lib/parse.js:8:15)
    at new LazyResult (/Users/stephen/Local Sites/test-1/app/public/wp-content/themes/understrap-child/node_modules/postcss/lib/lazy-result.js:133:16)
    at Processor.process (/Users/stephen/Local Sites/test-1/app/public/wp-content/themes/understrap-child/node_modules/postcss/lib/processor.js:28:14)
    at file:///Users/stephen/Local%20Sites/test-1/app/public/wp-content/themes/understrap-child/node_modules/postcss-cli/index.js:248:10
    at async Promise.all (index 1)
ERROR: "css-postcss" exited with 1.
ERROR: "css" exited with 1.
[nodemon] app crashed - waiting for file changes before starting...`

In the css directory it looks like the child-theme.css is being written, but not child-theme.min.css
Nothing in my system had changed, I hadnt ran any updates from friday to monday and so I am stumped why it stopped working.
Can anyone offer any steps to solve please?

**Comment by IanDelMar** [comment]
Have you tried to delete the node_modules folder, the child-theme.css, the child-theme.min.css and related sourcemaps and then to recompile the CSS files?

**Comment by shargreaves-millmead** [comment]
Seem to be getting somewhere - I had already tried deleting the node_modules and css files but that never done anything.
Have deleted (removed elsewhere for now) the src/build folder and it has now compiled after an error, but it seemed to run past it and compile and now seems ok.

The output it give me before compiling was this - (the build folder was still in place, but css files were not)

`> understrap-child@1.1.0 css-compile
> sass --style expanded --source-map --embed-sources --no-error-css --quiet src/sass/child-theme.scss:css/child-theme.css src/sass/custom-editor-style.scss:css/custom-editor-style.css

[!] Error: Could not resolve entry module (src/build/rollup.config.js).
Error: Could not resolve entry module (src/build/rollup.config.js).
    at error (/Users/stephen/Local Sites/test-1/app/public/wp-content/themes/understrap-child/node_modules/rollup/dist/shared/rollup.js:198:30)
    at ModuleLoader.loadEntryModule (/Users/stephen/Local Sites/test-1/app/public/wp-content/themes/understrap-child/node_modules/rollup/dist/shared/rollup.js:22306:20)
    at async Promise.all (index 0)

ERROR: "js-compile" exited with 1.
ERROR: "js" exited with 1.
[nodemon] app crashed - waiting for file changes before starting...`

It is now compiling and seems happy - will removing the build folder cause me problems down the line?

**Comment by IanDelMar** [comment]
This has to do with

> Have deleted (removed elsewhere for now) the src/build folder

Config files needed for the npm scripts to work are located at src/build.

**Comment by shargreaves-millmead** [comment]
Thank you for taking the time to help, I dont know how to troubleshoot npm so your help is greatly appreciated, as I know a little bit more than I did a couple of days ago.

It all seems to be running fine again - I removed the css files and the node_modules folder but that didnt do anything.
next step was that I removed the build folder, ran npm run watch, and it gave an error but went past the error and outputted the css files.

I have then added the build folder back in and put each file back into it one by one, running the script as I went and low and behold, no errors.

Running npm run watch, is compiling perfectly fine, npm run js is also working and outputting the child-theme.js

Thanks for your help, a mystery why it stopped working, but happy days as it is now working again.
